src/WebJobs.Script/Binding/FunctionBinding.cs

```diff
@@ -134,5 +121,13 @@ protected string Resolve(string name)
 
             return _config.HostConfig.NameResolver.ResolveWholeString(name);
         }
+
+        internal static void AddStorageAccountAttribute(Collection<CustomAttributeBuilder> attributes, string connection)
```

Comment:
The reason the core SDK attribute is binding type specific is by design. Those attributes can be applied at the parameter/function/class levels, so must be binding type specific.

Regarding the StorageFunctionBinding base class - I thought about that as well. I will do that.